beets/plugins.py

```diff
@@ -350,57 +351,55 @@ def load_plugins(names: Sequence[str] = ()) -> None:
     package in sys.path; the module indicated should contain the
     BeetsPlugin subclasses desired.
     """
+    classes = set()
+
     for name in names:
-        modname = f"{PLUGIN_NAMESPACE}.{name}"
         try:
-            try:
-                namespace = __import__(modname, None, None)
-            except ImportError as exc:
-                # Again, this is hacky:
-                if exc.args[0].endswith(" " + name):
-                    log.warning("** plugin {0} not found", name)
-                else:
-                    raise
-            else:
-                for obj in getattr(namespace, name).__dict__.values():
-                    if (
-                        isinstance(obj, type)
-                        and issubclass(obj, BeetsPlugin)
-                        and obj != BeetsPlugin
-                        and obj != MetadataSourcePlugin
-                        and obj not in _classes
-                    ):
-                        _classes.add(obj)
-
-        except Exception:
+            mod = import_module(f".{name}", package=PLUGIN_NAMESPACE)
+        except ModuleNotFoundError:
+            log.warning("** plugin {} not found", name)
+            continue
+        except ImportError:
             log.warning(
                 "** error loading plugin {}:\n{}",
                 name,
                 traceback.format_exc(),
             )
+            continue
 
+        for _name, cls in inspect.getmembers(mod, inspect.isclass):
+            if (
+                issubclass(cls, BeetsPlugin)
+                and cls != BeetsPlugin
+                and cls != MetadataSourcePlugin
+                and cls not in _instances
+            ):
+                classes.add(cls)
 
-_instances: dict[type[BeetsPlugin], BeetsPlugin] = {}
+    for cls in classes:
+        _register_plugin(cls)
+
+
+def _register_plugin(cls: type[BeetsPlugin]) -> None:
+    if cls in _instances:
+        return
+
```

Comment:
I doubt this check is required
```suggestion
```

Reply:
I think this also a left over from old code: We did use to do this repeatedly in the previous version of `find_plugins` (at least before the caching optimization was added).

If anything, it might make sense to log a warning here if a plugin is instantiated repeatedly. (Say if `load_plugins` is called repeatedly with the same plugins, which I don't think should happen.)